analyses/fusion_filtering/01-fusion-standardization.R

```diff
@@ -0,0 +1,90 @@
+# 
+# Command line arguments
+# fusionfile : 	Merged fusion calls from [STARfusion|| Arriba]
+#		For STARfusion default columns must be provided
+#		For Arriba along with default columns additional 
+#		columns from Fusion Annotator [fusion_name= GeneA--GeneB; 
+#		annots=Annotation from FusionAnnotator] are appended to the 
+#		right of original arriba data frame 
+#
+# caller :	Caller used for the input merged fusion calls [STARfusion|| Arriba]
+#
+
+library("optparse")
+
+option_list <- list( 
+     make_option(c("-f", "--fusionfile"),type="character",
+        help="Merged fusion calls from [STARfusion || Arriba]"),
+     make_option(c("-c", "--caller"), type="character", 
+        help="Caller type [STARfusion|| Arriba]"),
+     make_option(c("-o","--outputfile"),type="character",
+	help="Standardized fusion calls from [STARfusion || Arriba] (.RDS)")	
+    )
+
+# get command line options, if help option encountered print help and exit,
+
+opt <- parse_args(OptionParser(option_list=option_list))
+
+inputfile<-opt$fusionfile
+caller<-opt$caller
+outputfile<-opt$outputfile
+
+fusion_calls<-read.delim(opt$fusionfile,stringsAsFactors=FALSE)
+#To have a general column with unique IDs associated with each sample
+fusion_calls$Sample<-fusion_calls$tumor_id
+
+
+
+standard_fusion <- function(fusion_calls=fusion_calls,caller=caller) {
+# @param fusion_calls A dataframe from star fusion or arriba (more callers to be added)
+# @param caller string options STARfusion/arriba
+# @return Standardized fusion calls ready for filtering
+  
+  if(caller=="STARfusion"){
+  	fusion_calls$LeftBreakpoint <- gsub('^chr','',fusion_calls$LeftBreakpoint)
+  	fusion_calls$RightBreakpoint <- gsub('^chr','',fusion_calls$RightBreakpoint)
+  	fusion_calls$Fusion_Type<-fusion_calls$PROT_FUSION_TYPE
+  	fusion_calls$Fusion_Type<-gsub("INFRAME","in-frame",fusion_calls$Fusion_Type)
+  	fusion_calls$Fusion_Type<-gsub("FRAMESHIFT","frameshift",fusion_calls$Fusion_Type)
+  	fusion_calls$Fusion_Type[-which(fusion_calls$Fusion_Type %in% c("in-frame","frameshift"))] <- 'other'
+  	fusion_calls$Caller <- 'STARFusion'
+  	fusion_calls$FusionName<-fusion_calls$FusionName
+  	fusion_calls$Confidence<-"NA"
+  	standard_calls <- unique(fusion_calls[,c('FusionName','Sample','Caller','Fusion_Type','JunctionReadCount','SpanningFragCount','Confidence','annots')])
+  	return(standard_calls)
+
+  }
+  if( caller=="Arriba"){
+	fusion_calls$LeftBreakpoint <- gsub('^chr','',fusion_calls$breakpoint1)
+    	fusion_calls$RightBreakpoint <- gsub('^chr','',fusion_calls$breakpoint2)
+    	fusion_calls$Fusion_Type<-fusion_calls$reading_frame
+    	fusion_calls$Fusion_Type<-gsub("out-of-frame","frameshift",fusion_calls$Fusion_Type)
+    	fusion_calls$Fusion_Type[-which(fusion_calls$Fusion_Type %in% c("in-frame","frameshift"))] <- 'other'
+    	fusion_calls$Caller <- 'arriba'
+    	fusion_calls$FusionName <-paste0(gsub(",","/",fusion_calls$gene1),"--",gsub(",","/",fusion_calls$gene2))
```

Comment:
If I understand this correctly, you are changing the comma-separated gene identifiers to slash-separated gene identifiers here. If I check the `FusionName` column of the STARFusion output, there are no slashes. Can you comment on how this affects what you do downstream of standardization?

Reply:
Oh the "," are in Arriba calls if the fusion breakpoint in an intergenic region. The , separates the closest genes to the breakpoint

Reply:
Does STARFusion ever call a fusion breakpoint in an intergenic region? And/or do these always get filtered out? I'm essentially wondering if the choice of identifiers will cause problems in your "in _n_ callers" filtering later on.

Reply:
STARFusion only calls genic fusions so we don't find any issues with using the "/" here for intergenic  fusions since they are specific to arriba right now. But I get the concern here, other callers might have intergenic calls that might affect the "in n caller" part.

Reply:
Gotcha. No changes to be made here then, just something we should keep in mind if there are more callers added for this project.